test/DacpacTool.Tests/PackageBuilderTests.cs

```diff
@@ -463,6 +463,41 @@ public void ValidateModel_Warnings()
             result.ShouldBeTrue();
         }
 
+        [TestMethod]
+        public void ValidateModel_WarningsAsErrors()
+        {
+            // Arrange
+            var packageBuilder = new PackageBuilder();
+            packageBuilder.UsingVersion(SqlServerVersion.Sql150);
+            packageBuilder.TreatTSqlWarningsAsErrors = true;
+            packageBuilder.SetMetadata("MyPackage", "1.0.0.0");
+            packageBuilder.Model.AddObjects("CREATE PROCEDURE [csp_Test] AS BEGIN SELECT * FROM [dbo].[MyTable] END");
+
+            // Act
+            bool result = packageBuilder.ValidateModel();
+
+            // Assert
+            result.ShouldBeFalse();
```

Comment:
Perhaps we should add a comment here explaining why this isn't valid. It isn't immediately apparent based on the code.

Reply:
I'm not sure where this method is from, but you can write this as
```
result.Should().BeFalse("setting TreatTSqlWarningsAsErrors to true, SQL71502 Warning would be treated as error");
```

Reply:
@siriak This test method was fully copied from above code with some new calls. Base logic and code style of this method not changing